### PR TITLE
chore: optimize GitHub CI caching

### PR DESCRIPTION
**Pull request by sergeytimoshin**
Refactored workflows.

1. Fixed cache restoration in [setup-and-build workflow](https://github.com/Lightprotocol/light-protocol/pull/828/files#diff-a1315486761959ea28efdc433baee07a367ed6a1c57e37c8c1cfe6bb22a387a0).
2. Enabled [nx cloud caching](https://cloud.nx.app/orgs/651ec5e17fb0dbfbb6c2bd38/workspaces).
3. Fixed "out of disk space" situation by [removing unnecessary packages](https://github.com/Lightprotocol/light-protocol/pull/828/files#diff-a1315486761959ea28efdc433baee07a367ed6a1c57e37c8c1cfe6bb22a387a0R7-R13).
4. Switched back from buildjet to github ubuntu-latest vms.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -31,16 +31,25 @@ concurrency:
 jobs:
   test:
     if: github.event.pull_request.draft == false
-    runs-on: buildjet-4vcpu-ubuntu-2204
+    runs-on: ubuntu-latest
```
Comment (SwenSchaeferjohann) [inline .github/workflows/rust.yml:34]: fyi the github runner ran out of disk space in `rust.yml`, `js.yml`, `light-sdk-tests.yml`, `light-system-programs-test.yml`.

**Review by SwenSchaeferjohann — COMMENTED** [review]
@sergeytimoshin are the ci times measured here representative of using this optimized caching already?
I'm seeing some variance from 11 mins (js) to 27mins (forester), and the latter is less ideal

**Review by vadorovsky — APPROVED** [review]

**Comment by ananas-block** [comment]
@sergeytimoshin system program tests take 9min to build the CLI.
However, cli is only used in one e2e test and e2e functionality is already covered in the forester test.
How do you feel about moving the e2e tests that need cli to the forester just doing one round with everything enabled with probability 1 so that we test functionality.

Then we could remove CLI build from the system program tests.

**Comment by sergeytimoshin** [comment]
> @sergeytimoshin are the ci times measured here representative of using this optimized caching already? I'm seeing some variance from 11 mins (js) to 27mins (forester), and the latter is less ideal

27 mins was because of the missed cache, I tried to run it one more time and got 17 mins: https://github.com/Lightprotocol/light-protocol/actions/runs/9539487368/job/26334363586?pr=828

Generally I don't think we will experience here significant performance boost we switched back from buildjet-4cpus to github's ubuntu-latest vms, and github machines significantly slower.
And still, even with slower machines:
* forester: [18 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9530495899/job/26270429923) => [17 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9539487368/job/26334363586?pr=828)
* js tests: [26 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9551415181/job/26325668469) = > [11 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9539487365/job/26289891152?pr=828)
* light-sdk-tests: [19 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9551415195) => [11 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9539487359/job/26289890929?pr=828) 
* rust: [17 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9551415172) => [11 mins](https://github.com/Lightprotocol/light-protocol/actions/runs/9539487363/job/26289890680?pr=828)

**Pull request merged**